lab-taylor/app/controllers/game-controller.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+const angular = require('angular');
+const Monster = require('../model/monster');
+const Item = require('../model/items');
+
+angular.module('adventureGame').controller('GameController', [GameController]);
+
+function GameController() {
+  this.history = ['STATION REPORT -- ALL SYSTEMS DOWN -- GOOD LUCK'];
+
+  this.map = require('../model/map');
+  this.dynamicClass = '';
+  this.player = {
+    name: 'Reptar',
+    hp: 100,
+    damage: 5,
+    xp: 0,
+    location: 'moduleA1',
+    items: []
+  };
+
+  this.gameForm = {
+    direction: 'forward'
+  };
+
+  this.stationModule = {
+    name: this.player.location
+  };
+
+  this.moveCount = 1;
+}
+
+GameController.prototype.moveDirection = function (direction) {
+  this.moveCount++;
+  const oldLocation = this.player.location;
+  const newLocation = this.map[oldLocation][direction];
+
+  if (newLocation && newLocation !== 'hull') {
+    this.updateLocation(newLocation);
+    return;
+  }
+
+  this.holdLocation();
+  return;
+};
+
+GameController.prototype.updateLocation = function(location) {
+  this.player.location = location;
+  this.stationModule.name = location;
+  this.stationModule.item = null;
+
+  if (Math.random() < this.map[location].monsterChance) {
+    this.stationModule.monster = new Monster();
+    this.player.hp -= this.stationModule.monster.damage;
+    this.updateHistory(`ran into ${this.stationModule.monster.name} and lost ${this.stationModule.monster.damage} hp`);
+    if (this.amIDead()) this.dead();
+    return;
+  }
+
+  if (Math.random() < this.map[location].itemChance) {
+    this.stationModule.item = new Item();
+    this.pickUpItem(this.stationModule.item.name);
+  }
+
+  this.stationModule.monster = null;
+  this.stationModule.item = null;
+  this.updateHistory(`is now in ${this.player.location}.`);
+};
+
+GameController.prototype.attackMonster = function() {
+  if (this.stationModule.monster) {
+    this.moveCount++;
+    let message = '';
+
+    this.stationModule.monster.hp -= this.player.damage;
+    this.updateHistory(`did ${this.player.damage} damage to the monster.`);
+    if (Math.random() > 0.5) {
+      this.player.hp -= this.stationModule.monster.damage;
+      message += ` -- the monster hurt you! You lost ${this.stationModule.monster.damage} hp.`;
+      this.updateHistory(message);
+    }
+
+    if(this.stationModule.monster.hp < 0) {
```

Comment:
I would change this to <= 0. I had a monster at 0 and it didn't register as dead.